KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java

```diff
@@ -732,41 +732,36 @@ else if (canBuildRoad(roadNodes.get(i))) {
 
                         //check that the settlement is unowned
                         if (settlementNodes.get(i).getPlayer() == 0) {
-                            //check what mode the game is in 
-                            if (inSetup && playerSetupSettlementLeft > 0) { // In Setup
-                                // Place one of the player's setup settlements
-                                settlementNodes.get(i).setPlayer(currentPlayer);
-                                playerSetupSettlementLeft--;
-                                // Increment the player's victory point counter
-                                victoryPoints[currentPlayer]++;
-                              
-                                // Update thwe build buttons to relfect the remaining setup buildings
-                                updateBuildButtons();
-                            } // If the main game is in progress and the player can build the settlement there
-                            else if (canBuildSettlement(settlementNodes.get(i))) {
-                                // The card check has already been made, and the user has the right cards
-                                
-                                // Remove the cards from the player's deck
-                                // Remove 1 clay, 1 wood, 1 wheat, and 1 sheep
-                                cards[currentPlayer].remove(new Integer(1));
-                                cards[currentPlayer].remove(new Integer(2));
-                                cards[currentPlayer].remove(new Integer(3));
-                                cards[currentPlayer].remove(new Integer(4));
 
+                            // Check that the player can build there and update the instruction labels accordingly if hey cannot
+                            if (canBuildSettlement(settlementNodes.get(i))) {
```

Comment:
This function now checks if setup buildings are placed too close as well, so i modified settlement building to use parallel structure